common/testutils/test_utils.go

```diff
@@ -0,0 +1,74 @@
+package testutils
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/exp/rand"
+	"testing"
+	"time"
+)
+
+// InitializeRandom initializes the random number generator. Prints the seed so that the test can be rerun
+// deterministically. Replace a call to this method with a call to initializeRandomWithSeed to rerun a test
+// with a specific seed.
+func InitializeRandom() {
+	rand.Seed(uint64(time.Now().UnixNano()))
+	seed := rand.Uint64()
+	fmt.Printf("Random seed: %d\n", seed)
+	rand.Seed(seed)
+}
+
+// InitializeRandomWithSeed initializes the random number generator with a specific seed.
+func InitializeRandomWithSeed(seed uint64) {
+	fmt.Printf("Random seed: %d\n", seed)
+	rand.Seed(seed)
+}
+
+// AssertEventuallyTrue asserts that a condition is true within a given duration. Repeatably checks the condition.
+func AssertEventuallyTrue(t *testing.T, condition func() bool, duration time.Duration, debugInfo ...any) {
+	start := time.Now()
+	for time.Since(start) < duration {
```

Comment:
Could we use a channel/a context/select statement to achieve the same thing instead of polling every millisecond?

Reply:
I've refactored this to use a select clause. Still learning best practices, please let me know if it needs more tweaks.